### ISSUE TITLE
Fix definition of GroupNormalization

### ISSUE THREAD

**Issue by rocking5566** [onnx/onnx#5466]
# Ask a Question

### Question
According to the document of GroupNormalization's operator 
https://github.com/onnx/onnx/blob/main/docs/Operators.md#groupnormalization
I found the tensor shape of both scale and bias are num_groups.
However, this should be equal to C instead of G.

### Further information
- Relevant Area: operators
-According to the paper (page 4)
https://arxiv.org/pdf/1803.08494.pdf
Shape of Gamma (scale) and beta (bias) ic C (perchannel not pergroup).
![image](https://github.com/onnx/onnx/assets/9115697/8622ff58-122f-4ee4-a182-9f7f14caecc3)

**Comment by gramalingam** [comment]
@yuanyao-nv , any thoughts on this?

**Comment by yuanyao-nv** [comment]
@rocking5566 Thanks for bringing this up. It does seem like the ONNX spec is being more restrictive than the version in the paper and PyTorch.
The current ONNXRuntime implementation, though, follows the ONNX spec with per-group scale and bias. What you linked to above seems to be the `com.microsoft.GroupNorm` op. 

To determine next steps, I think it would be useful to look at some practical use cases first. I'm wondering if you have some concrete models where per-channel scale and bias are needed?

**Comment by rocking5566** [comment]
> @rocking5566 Thanks for bringing this up. It does seem like the ONNX spec is being more restrictive than the version in the paper and PyTorch. The current ONNXRuntime implementation, though, follows the ONNX spec with per-group scale and bias. What you linked to above seems to be the `com.microsoft.GroupNorm` op.
> 
> To determine next steps, I think it would be useful to look at some practical use cases first. I'm wondering if you have some concrete models where per-channel scale and bias are needed?

Stable diffusion require groupnorm operator.
You could reference follow model
https://github.com/microsoft/onnxruntime/blob/main/onnxruntime/python/tools/transformers/models/stable_diffusion/README.md

Actually, I am working on AMD rocm team and provide groupnorm backend for above model.
I imeplement the per-channel scale and bias and provide rocm backend to onnxruntime.
However, I found onnx spec is different from other framework.

By the way, you mention the link I provide is **com.microsoft.GroupNorm**.  I wonder is there any other groupnorm in onnxruntime is per-group scale and bias?

**Comment by yuanyao-nv** [comment]
@rocking5566 Thanks for the reference. Does this stable diffusion model actually use different scale and bias for each channel? Or do it use the default values of 1 and 0 respectively? I've seen resnet models where the scale and bias are always kept at the default values. But regardless, I think matching pytorch's implementaion is strong enough reason for us to extend this op in coming opsets. I'm just trying to gather more info on the impact of this limitation.

In onnxruntime, the standard implementation for `GroupNormalization` (note the name difference) uses per-group sclae and bias. You can verify this by running a simple onnx model through onnxruntime. If the scale and bias inputs have length C instead of G, then it'll complain about shape mismatch.

**Comment by rocking5566** [comment]
> GroupNormalization

stable diffusion does not use default value of gamma and beta.
In addition, traditional Resnet use batchnorm instead of groupnorm.

**Comment by gramalingam** [comment]
This was discussed in the operator SIG meeting last week, and the op's spec will be updated.

**Comment by justinchuby** [comment]
Should we deprecate the old spec?

**Comment by justinchuby** [comment]
@yuanyao-nv gentle nudge. Do you have any updates? We are targeting this for the 1.16 release.

**Comment by yuanyao-nv** [comment]
> @yuanyao-nv gentle nudge. Do you have any updates? We are targeting this for the 1.16 release.

As I mentioned in my message to @gramalingam , I wonder if we should apply an individual fix to this op or should we make this fix when unifying the normalization ops. An individual fix should be faster if we are targeting the next release.

**Comment by justinchuby** [comment]
I vote for fixing this op first, as the error has been around for a while and it is preventing adoption of this operator.

**Comment by yuanyao-nv** [comment]
@justinchuby May I ask what's the projected timeline for 1.16 release?

**Comment by justinchuby** [comment]
@gramalingam could you chime in? Thanks!
